beacon-chain/state/v3/state_trie.go

```diff
@@ -0,0 +1,71 @@
+package v3
```

Comment:
This one has a huge logic extra from v2 from the balance trie branch, it can be copied and pasted. cc @nisdas  We should merge these PRs faster to avoid this happening

Reply:
`rootSelector` will be implemented in the follow up PR